ui/guest/src/contentController.ts

```diff
@@ -379,12 +380,37 @@ export function duplicateItem(modelId: string, fieldId: string, index: number |
 export function insertItem(modelId: string, fieldId: string, index: number | string, contentType: ContentType): void {
   const instance: InstanceRecord = {};
   const models = getCachedModels();
-  Object.entries(contentType.fields[fieldId].fields).forEach(([id, field]) => {
+  const fields = fieldId.split('.');
+  const field = fields.reduce((acc, field) => acc[field].fields, contentType.fields);
+  // For each of the entries of the field of type `content-type`, set the default value for the new instance.
+  Object.entries(field).forEach(([id, field]) => {
     if (!systemProps.includes(field.id)) {
       instance[id] = getDefaultValue(field);
     }
   });
 
+  const indexPieces = index?.toString().split('.');
+  const concatFieldId = !isSimple(fieldId)
+    ? mergeArraysAlternatively(fieldId.split('.'), indexPieces).join('.')
+    : fieldId;
+  const currentItems = getCollection(models[modelId], concatFieldId, index) ?? [];
+
+  let newItems;
+  if (nou(index) || indexPieces.length < fields.length) {
+    // Index doesn't exist for current field subsection (when not simple field), or index doesn't exist at all, insert at the end
+    newItems = [...currentItems, instance];
+  } else {
+    // If index is complex, we already have the set of items from the model, now we need the latter part of the index
+    // to insert the new instance at the desired position.
+    let targetIndex = isSimple(index) ? index : popPiece(index as string);
+    newItems = currentItems.splice(targetIndex as number, 0, instance);
+  }
+  const model = setCollection(models[modelId], fieldId, index, newItems);
```

Comment:
Do not mutate the existing instance inside of models.

Reply:
The additional complexity here is that shouldn't mutate the inner object target of the collection being set